### PR TITLE
Dialogflow detect intent streams

### PR DESCRIPTION
**Pull request by telpirion**
This PR adds the following code sample region tags:
dialogflow_detect_intent_streaming

The canonical (Java) is found here:
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/dialogflow/cloud-client/src/main/java/com/example/dialogflow/DetectIntentStream.java

**Note**:  The Dialogflow tests seem to be disabled. ([See commented out lines](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/master/dialogflow/api/Test/runTests.ps1#L21)).

### CODE REVIEW

File: dialogflow/api/Test/DetectIntentStreamTest.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (amanda-tarafa) [inline dialogflow/api/Test/DetectIntentStreamTest.cs]: License missing.

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (telpirion) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: Done

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using CommandLine;
```
[the remark this answers is not in the document]
Comment (telpirion) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: Done

**Comment by amanda-tarafa** [comment]
FYI:
I've created #936 to reactivate Dialogflow tests. Let's wait till that's in so that we can make sure that this PR still passes.

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: License should be at the top of the file.

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                            Console.Write("Intent detected:");
+                            Console.WriteLine(queryResult.Intent.DisplayName);
+                        }
+                    }
+                }
+            });
+
+            // Instructs the speech recognizer how to process the audio content.
+            // Note: hard coding audioEncoding, sampleRateHertz for simplicity.
+            // Audio encoding of the audio content sent in the query request.
```
Comment (amanda-tarafa) [inline dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: nit: I believe this line is out of place now.

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+        [Verb("detect-intent:streams", HelpText = "Detect intent from stream")]
+        public class DetectIntentFromStreamOptions : OptionsWithProjectIdAndSessionId
+        {
+            [Value(0, MetaName = "file", HelpText = "Path to the audio file", Required = true)]
+            public string FilePath { get; set; }
+        }
+
+        // [START dialogflow_detect_intent_streaming]
+        public static async Task<object> DetectIntentFromStream(string projectId,
```
[the remark this answers is not in the document]
Comment (telpirion) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: Done

File: dialogflow/api/Test/DetectIntentStreamTest.cs (hunk 1 above)
Comment (telpirion) [reply dialogflow/api/Test/DetectIntentStreamTest.cs]: Done

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs (hunk 4 above)
Comment (telpirion) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: I believe that the comment correctly describes lines 76-84. The QueryInput object includes the configuration files that the speech recognizer will use when recognizing the audio data. The configuration hard codes several settings that developers will need to change for their own data.

Let me know if I'm missing something.

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (telpirion) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: Fixed.

File: dialogflow/api/DialogflowSamples/DetectIntentStream.cs (hunk 4 above)
Comment (amanda-tarafa) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: I meant just the last line, that seems better on top of line 80. But, yeah, it's really not bad if here, leave it as is.

Comment (telpirion) [reply dialogflow/api/DialogflowSamples/DetectIntentStream.cs]: Fair enough, I think that line is probably extraneous. Done.

**Pull request merged**